spark-hpcc/src/main/java/org/hpccsystems/spark/HpccFile.java

```diff
@@ -166,6 +168,35 @@ public HpccRDD getRDD(SparkContext sc) throws HpccFileException
     rdd.setTraceContext(parentTraceID, parentSpanID);
     return rdd;
   }
+
+  /**
+   * Make a Spark Resilient Distributed Dataset (RDD) that provides access
+   * to THOR based datasets.
+   * @param sc Spark Context
+   * @return An RDD of THOR data.
+   * @throws HpccFileException When there are errors reaching the THOR data
+   */
+  public HpccRDD getRDD(SparkContext sc, List<Integer> filePartsToRead) throws HpccFileException
+  {
+    DataPartition[] fileParts = getFileParts();
+    
+    List<DataPartition> filteredFileParts = new ArrayList<DataPartition>();
+    for (Integer filePart : filePartsToRead)
+    {
+      filePart--; // File parts are 1 based
+      if (filePart < 0 || filePart >= fileParts.length)
+      {
+        throw new HpccFileException("Invalid file part requested: " + filePart);
+      }
+
+      filteredFileParts.add(fileParts[filePart]);
+    }
+
+	  HpccRDD rdd = new HpccRDD(sc, filteredFileParts.toArray(new DataPartition[0]), this.getRecordDefinition(), this.getProjectedRecordDefinition(), this.getFileAccessExpirySecs(), this.recordLimit);
```

Comment:
It's not clear to me what this syntax implies:
toArray(new DataPartition[0])

Reply:
Yeah, that syntax is sort of weird, List.toArray() uses the type of the array passed in to know the correct return type.